mesonbuild/compilers/compilers.py

```diff
@@ -1327,8 +1305,9 @@ def get_pic_args(self):
     def get_buildtype_args(self, buildtype):
         return gnulike_buildtype_args[buildtype]
 
+    @abc.abstractmethod
     def get_optimization_args(self, optimization_level):
-        return gnu_optimization_args[optimization_level]
+        return
```

Comment:
Shouldn't we raise an assert in the abstract method implementation [in case someone does `super()` to call it](https://docs.python.org/3/library/abc.html#abc.abstractmethod)? Or does just having `return` not count as an implementation?

Reply:
Done